backend/repositories/GameRepository.ts

```diff
@@ -0,0 +1,25 @@
+import { GameData } from '../types/Game'
+import { InputError } from '../errors/InputError'
+import * as storage from '../storage/Storage'
+
+const isValidGame = (gameData: GameData): boolean => {
+  if (!gameData.name || !gameData.description) {
+    return false
+  } else if (gameData.name.trim() != gameData.name) {
+    return false
+  } else if (gameData.name.toLowerCase() != gameData.name) {
+    return false
+  } else if (gameData.name.includes(' ')) {
+    return false
+  } else if (gameData.description.trim() != gameData.description) {
+    return false
+  }
+  return true
+}
+
+export const addGame = async (gameData: GameData): Promise<void> => {
+  if (!isValidGame(gameData)) {
```

Comment:
why are you allowing a creation of an invalid game? why not validate it right at the time of construction? (ie https://github.com/salsita/foosball-rating/pull/113/files#diff-382e120a6484790f7829e0ec3bb18713R9)

Reply:
GameData is just description of the object created from body of the request - it's created by express.

Reply:
hmm then this is an antipattern. you're supplying any into addGame - you don't know if the argument has a given type.

here's what I'd suggest:

```

interface GameData {
}

// notice the test IS GameData as a return postcondition
function isGameData(test: unknown): test is GameData {
    // use the conditions from your isValidGame
    // if this function returns true is means that the tested variable has the GameDataType
    return "the conditions from your in isValidGame" != test;
}

function addGame(maybeGameData: unknown) {
  if (!isGameData(maybeGameData)) {
    throw Error("Not game info!")
  }

  // insert into storage
}



```

Reply:
more can be found here https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates

Reply:
also, it's weird that the TS compiler didn't protest against you supplying any. maybe i'd tighten the screws, so to speak (you can have it as a separate issue in the future - https://stackoverflow.com/questions/32227277/typescript-typed-function-argument-accepts-any )

Reply:
antipattern fixed in https://github.com/salsita/foosball-rating/pull/113/commits/f657e38aac9173ce08efb3ec3cf4c740fc7b689b

Reply:
issue filed for 'any' https://github.com/salsita/foosball-rating/issues/133